internal/migrations/20201215140033_non_nil_disk_eligibility.go

```diff
@@ -0,0 +1,94 @@
+package migrations
+
+import (
+	"encoding/json"
+	"github.com/jinzhu/gorm"
+	"github.com/openshift/assisted-service/models"
+	gormigrate "gopkg.in/gormigrate.v1"
+)
+
+// migrateHostInventory behaves similarly to (m *host.Manager) populateDisksEligibility in the sense that
+// it sets all nil eligibility structs to "eligible".
+func migrateHostInventory(inventoryString string, rollback bool) (string, error) {
+	var inventory models.Inventory
+	if err := json.Unmarshal([]byte(inventoryString), &inventory); err != nil {
+		return "", err
+	}
+
+	for _, disk := range inventory.Disks {
+		if !rollback {
+			if disk.InstallationEligibility == nil {
+				// This is a sign that the agent is using an old version that does not fill in the eligibility struct,
+				// for backwards compatibility, pretend that the agent has decided that this disk is eligible
+				disk.InstallationEligibility = &models.DiskInstallationEligibility{Eligible: true,
+					NotEligibleReasons: make([]string, 0)}
+			}
+		} else {
+			disk.InstallationEligibility = nil
+		}
+	}
+
+	result, err := json.Marshal(inventory)
+	if err != nil {
+		return "", err
+	}
+
+	return string(result), nil
+}
+
+func migrateCluster(tx *gorm.DB, cluster *models.Cluster, rollback bool) error {
+	if err := tx.Preload("Hosts").First(&cluster, "id = ?", cluster.ID).Error; err != nil {
+		return err
+	}
+
+	for _, host := range cluster.Hosts {
+		var err error
+		host.Inventory, err = migrateHostInventory(host.Inventory, rollback)
+		if err != nil {
+			return err
+		}
+
+		if err := tx.Model(&host).Update("inventory", host.Inventory).Error; err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func migrateClusters(tx *gorm.DB, rollback bool) error {
+	clusters, err := tx.Model(&models.Cluster{}).Rows()
```

Comment:
Why even query for clusters? Looks like we should be querying the hosts directly.
Maybe in batches to prevent pulling all the hosts into memory at once.

Reply:
Good point. Initially I thought "For each cluster, I need to modify all hosts in that cluster" but it does seem like a useless indirection, now that you mention it.